Turn on thread-safe statics everywhere except bmalloc and JSC

cdumez opened the pull request:
#### 2d4ca326601eb0a377673f8200349913aae83c65
<pre>
Turn on thread-safe statics everywhere except bmalloc and JSC
<a href="https://bugs.webkit.org/show_bug.cgi?id=303020">https://bugs.webkit.org/show_bug.cgi?id=303020</a>

Reviewed by Geoffrey Garen.

Turn on thread-safe statics everywhere except bmalloc and JSC. This
results in safer &amp; simpler code without any noticeable performance
regression on the benchmarks we track.

* Configurations/CommonBase.xcconfig:
* Source/WTF/Configurations/Base.xcconfig:
* Source/WTF/wtf/Assertions.cpp:
* Source/WTF/wtf/CryptographicallyRandomNumber.cpp:
* Source/WTF/wtf/CurrentTime.cpp:
* Source/WTF/wtf/FastMalloc.cpp:
(WTF::MallocCallTracker::singleton):
* Source/WTF/wtf/Language.cpp:
(WTF::observerMap):
* Source/WTF/wtf/Logger.cpp:
(WTF::Logger::observers):
(WTF::Logger::messageHandlerObservers):
* Source/WTF/wtf/MainThreadDispatcher.cpp:
(WTF::MainThreadDispatcher::singleton):
* Source/WTF/wtf/MemoryPressureHandler.cpp:
(WTF::MemoryPressureHandler::singleton):
* Source/WTF/wtf/MemoryPressureHandler.h:
* Source/WTF/wtf/NumberOfCores.cpp:
(WTF::numberOfPhysicalProcessorCores):
* Source/WTF/wtf/OSRandomSource.cpp:
(WTF::cryptographicallyRandomValuesFromOS):
* Source/WTF/wtf/ParkingLot.cpp:
* Source/WTF/wtf/RAMSize.cpp:
(WTF::ramSize):
* Source/WTF/wtf/RunLoop.cpp:
(WTF::RunLoop::runLoopHolder):
* Source/WTF/wtf/SequesteredImmortalHeap.cpp:
(WTF::SequesteredImmortalHeap::instance):
* Source/WTF/wtf/StatisticsManager.cpp:
(WTF::StatisticsManager::singleton):
(WTF::StatisticsManager::~StatisticsManager):
* Source/WTF/wtf/StatisticsManager.h:
* Source/WTF/wtf/Threading.cpp:
(WTF::Thread::allThreads):
(WTF::Thread::initializeInThread):
* Source/WTF/wtf/TranslatedProcess.cpp:
(WTF::isX86BinaryRunningOnARM):
* Source/WTF/wtf/URL.cpp:
(WTF::aboutBlankURL):
(WTF::aboutSrcDocURL):
* Source/WTF/wtf/URLParser.cpp:
(WTF::URLParser::internationalDomainNameTranscoder):
* Source/WTF/wtf/UUID.cpp:
(WTF::bootSessionUUIDString):
* Source/WTF/wtf/WorkQueue.cpp:
(WTF::ConcurrentWorkQueue::apply):
(WTF::WorkQueue::mainSingleton):
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.mm:
(WTF::CocoaApplication::shouldOSFaultLogForAppleApplicationUsingWebKit1):
* Source/WTF/wtf/cocoa/SystemTracingCocoa.cpp:
(WTFSignpostLogHandle):
* Source/WTF/wtf/text/TextBreakIterator.cpp:
(WTF::TextBreakIteratorCache::singleton):
* Source/WTF/wtf/text/TextBreakIterator.h:
* Source/WTF/wtf/threads/Signals.cpp:
(WTF::activeThreads):
* Source/WTF/wtf/unicode/icu/CollatorICU.cpp:
(WTF::resolveDefaultLocale):
* Source/WebCore/Configurations/Base.xcconfig:
* Source/WebCore/Modules/applepay/ApplePayLogoSystemImage.mm:
(WebCore::passKitBundleSingleton):
(WebCore::applePayLogoWhite):
(WebCore::applePayLogoBlack):
* Source/WebCore/Modules/mediastream/RTCDataChannel.cpp:
(WebCore::WTF_REQUIRES_LOCK):
* Source/WebCore/Modules/system-preview/ARKitBadgeSystemImage.mm:
(WebCore::arKitBundle):
(WebCore::systemPreviewLogo):
* Source/WebCore/PAL/Configurations/Base.xcconfig:
* Source/WebCore/PAL/pal/cocoa/AVFoundationSoftLink.mm:
(PAL::AVFoundationLibrary):
* Source/WebCore/PAL/pal/system/ios/Device.cpp:
(PAL::deviceName):
* Source/WebCore/PAL/pal/text/EncodingTables.cpp:
(PAL::jis0208):
(PAL::jis0212):
(PAL::big5):
(PAL::gb18030):
* Source/WebCore/PAL/pal/text/TextCodecCJK.cpp:
(PAL::jis0208EncodeIndex):
(PAL::eucKREncodingIndex):
(PAL::big5EncodeIndex):
(PAL::gb18030EncodeIndex):
* Source/WebCore/PAL/pal/text/TextCodecSingleByte.cpp:
(PAL::tableForEncoding):
* Source/WebCore/SaferCPPExpectations/RetainPtrCtorAdoptCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLambdaCapturesCheckerExpectations:
* Source/WebCore/bindings/js/WebCoreJSClientData.cpp:
(WebCore::JSHeapData::ensureHeapData):
* Source/WebCore/crypto/CryptoAlgorithmRegistry.cpp:
(WebCore::CryptoAlgorithmRegistry::singleton):
* Source/WebCore/crypto/CryptoAlgorithmRegistry.h:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::maskedURLForBindings):
* Source/WebCore/editing/cocoa/DictionaryLookup.mm:
(WebCore::canCreateRevealItems):
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
(WebCore::activeContexts):
* Source/WebCore/html/track/TrackBase.cpp:
(WebCore::nullLogger):
* Source/WebCore/loader/ResourceMonitorThrottlerHolder.cpp:
(WebCore::ResourceMonitorThrottlerHolder::sharedWorkQueueSingleton):
* Source/WebCore/loader/cache/CachedResource.cpp:
(WebCore::CachedResource::response const):
(WebCore::CachedResource::resourceError const):
* Source/WebCore/page/NavigatorBase.cpp:
(WebCore::NavigatorBase::platform const):
(WebCore::NavigatorBase::hardwareConcurrency):
* Source/WebCore/page/NavigatorUAData.cpp:
(WebCore::NavigatorUAData::platform const):
* Source/WebCore/page/Performance.cpp:
(WebCore::isSignpostEnabled):
* Source/WebCore/page/cocoa/ResourceUsageOverlayCocoa.mm:
(WebCore::historicUsageData):
* Source/WebCore/page/linux/ResourceUsageThreadLinux.cpp:
(WebCore::threadInfoMap):
* Source/WebCore/page/scrolling/ScrollingThread.cpp:
(WebCore::ScrollingThread::singleton):
* Source/WebCore/page/scrolling/ScrollingThread.h:
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::webCoreBundleSingleton):
* Source/WebCore/platform/PublicSuffixStore.cpp:
(WebCore::PublicSuffixStore::singleton):
* Source/WebCore/platform/PublicSuffixStore.h:
* Source/WebCore/platform/cocoa/WebAVPlayerLayerView.mm:
(WebCore::allocWebAVPlayerLayerViewInstance):
(WebCore::allocWebAVPictureInPicturePlayerLayerViewInstance):
* Source/WebCore/platform/graphics/DestinationColorSpace.cpp:
(WebCore::knownColorSpace):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::globalLoaderDelegateQueue):
(WebCore::MediaPlayerPrivateAVFoundationObjC::asyncVideoPlaybackQualityMetrics):
* Source/WebCore/platform/graphics/avfoundation/objc/QueuedVideoOutput.mm:
(WebCore::globalOutputDelegateQueue):
* Source/WebCore/platform/graphics/cg/ColorCG.cpp:
(WebCore::cachedCGColorTransform):
(WebCore::cachedCGColor):
* Source/WebCore/platform/graphics/cocoa/WebActionDisablingCALayerDelegate.mm:
(+[WebActionDisablingCALayerDelegate shared]):
* Source/WebCore/platform/graphics/cocoa/WebMAudioUtilitiesCocoa.mm:
(WebCore::registerOpusDecoderIfNeeded):
* Source/WebCore/platform/graphics/gpu/cocoa/GPUAvailabilityMetal.mm:
(WebCore::isGPUAvailable):
* Source/WebCore/platform/ios/PlatformScreenIOS.mm:
(WebCore::screenPPIFactor):
* Source/WebCore/platform/ios/ValidationBubbleIOS.mm:
(allocWebValidationBubbleViewControllerInstance):
* Source/WebCore/platform/ios/VideoPresentationInterfaceAVKitLegacy.mm:
(allocWebAVPictureInPictureContentViewControllerInstance):
* Source/WebCore/platform/ios/WebItemProviderPasteboard.mm:
(+[WebItemProviderPasteboard sharedInstance]):
* Source/WebCore/workers/WorkerOrWorkletThread.cpp:
(WebCore::WorkerOrWorkletThread::workerOrWorkletThreads):
* Source/WebGPU/Configurations/Base.xcconfig:
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::Device::nopVertexFunction):
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
(WebGPU::ComputePassEncoder::runPredispatchIndirectCallValidation):
* Source/WebGPU/WebGPU/Device.mm:
* Source/WebGPU/WebGPU/HardwareCapabilities.mm:
(WebGPU::isShaderValidationEnabled):
(WebGPU::isPhysicalHardware):
* Source/WebInspectorUI/Configurations/Base.xcconfig:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.cpp:
(WebKit::remoteGraphicsContextGLStreamWorkQueueSingleton):
* Source/WebKit/GPUProcess/graphics/wc/RemoteWCLayerTreeHost.cpp:
(WebKit::remoteGraphicsStreamWorkQueue):
* Source/WebKit/GPUProcess/media/RemoteMediaResourceLoader.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
(WebKit::NetworkRTCProvider::rtcNetworkThread):
(WebKit::NetworkRTCProvider::createSocket): Deleted.
(WebKit::NetworkRTCProvider::signalSocketIsClosed): Deleted.
(WebKit::NetworkRTCProvider::protectedRTCMonitor): Deleted.
(WebKit::NetworkRTCProvider::sharedPreferencesForWebProcess): Deleted.
(WebKit::NetworkRTCProvider::updateSharedPreferencesForWebProcess): Deleted.
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCTCPSocketCocoa.mm:
(WebKit::tcpSocketQueueSingleton):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.mm:
(WebKit::udpSocketQueueSingleton):
* Source/WebKit/Platform/IPC/cocoa/ConnectionCocoa.mm:
(IPC::shouldLogIncomingMessageHandling):
* Source/WebKit/Platform/classifier/cocoa/ResourceLoadStatisticsClassifierCocoa.cpp:
(WebKit::ResourceLoadStatisticsClassifierCocoa::singletonPredictionModel):
* Source/WebKit/Platform/cocoa/ImageAnalysisUtilities.mm:
(WebKit::shouldLogFullImageTranslationResults):
* Source/WebKit/Shared/API/APIObject.cpp:
(API::Object::apiObjectsUnderConstruction):
* Source/WebKit/Shared/Cocoa/CompletionHandlerCallChecker.mm:
(WebKit::shouldThrowExceptionForDuplicateCompletionHandlerCall):
* Source/WebKit/Shared/Cocoa/CoreIPCSecureCoding.mm:
(WebKit::SecureCoding::internalClassNamesExemptFromSecureCodingCrash):
* Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm:
(WebKit::isInWebKitChildProcess):
(WebKit::doesParentProcessHaveTrackingPreventionEnabled):
(WebKit::isParentProcessAFullWebBrowser):
(WebKit::isFullWebBrowserOrRunningTest):
* Source/WebKit/Shared/Cocoa/LaunchLogHook.h:
* Source/WebKit/Shared/Cocoa/LaunchLogHook.mm:
(WebKit::LaunchLogHook::singleton):
* Source/WebKit/Shared/Cocoa/WebPreferencesDefaultValuesCocoa.mm:
(WebKit::defaultTextRecognitionInVideosEnabled):
(WebKit::defaultVisualTranslationEnabled):
(WebKit::defaultRemoveBackgroundEnabled):
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.mm:
(WebKit::setOSTransaction):
* Source/WebKit/Shared/Extensions/WebExtensionUtilities.mm:
(WebKit::classToClassString):
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultMediaSessionCoordinatorEnabled):
(WebKit::defaultDigitalCredentialsEnabled):
* Source/WebKit/Shared/ios/WebPreferencesDefaultValuesIOS.mm:
(WebKit::isAsyncTextInputFeatureFlagEnabled):
* Source/WebKit/Shared/unix/BreakpadExceptionHandler.cpp:
(WebKit::installBreakpadExceptionHandler):
* Source/WebKit/UIProcess/API/Cocoa/APIContentRuleListStoreCocoa.mm:
(API::ContentRuleListStore::defaultStorePath):
* Source/WebKit/UIProcess/API/Cocoa/WKNSURLAuthenticationChallenge.mm:
(-[WKNSURLAuthenticationChallenge _web_createTarget]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:
(+[WKWebsiteDataStore allWebsiteDataTypes]):
(+[WKWebsiteDataStore _allWebsiteDataTypesIncludingPrivate]):
* Source/WebKit/UIProcess/Cocoa/MediaPermissionUtilities.mm:
(WebKit::checkSandboxRequirementForType):
(WebKit::checkUsageDescriptionStringForType):
* Source/WebKit/UIProcess/Cocoa/UIProcessLogInitializationCocoa.mm:
(WebKit::UIProcess::wtfLogLevelString):
(WebKit::UIProcess::webCoreLogLevelString):
(WebKit::UIProcess::webKitLogLevelString):
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:
(WebKit::WebKitSwiftLibrary):
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::setMediaAccessibilityPreferences):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCommandCocoa.mm:
(WebKit::WebExtensionCommand::setActivationKey):
* Source/WebKit/UIProcess/Extensions/Cocoa/_WKWebExtensionDeclarativeNetRequestRule.mm:
(validRequestMethodsString):
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::ProcessLauncher::hasExtensionsInAppBundle):
* Source/WebKit/UIProcess/Media/cocoa/MediaUsageManagerCocoa.mm:
(WebKit::usageTrackingAvailable):
* Source/WebKit/UIProcess/Notifications/glib/NotificationService.cpp:
(WebKit::NotificationService::singleton):
* Source/WebKit/UIProcess/Notifications/glib/NotificationService.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeViews.mm:
(WebKit::scrollViewScrollIndicatorClassSingleton):
* Source/WebKit/UIProcess/SuspendedPageProxy.cpp:
(WebKit::messageNamesToIgnoreWhileSuspended):
* Source/WebKit/UIProcess/WebPasteboardProxy.cpp:
(WebKit::WebPasteboardProxy::singleton):
* Source/WebKit/UIProcess/WebPasteboardProxy.h:
* Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm:
(WebKit::defaultWebsiteDataStoreRootDirectory):
(WebKit::WebsiteDataStore::tempDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::cacheDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::websiteDataDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::defaultResolvedContainerTemporaryDirectory):
* Source/WebKit/UIProcess/XR/ios/PlatformXRSystemIOS.mm:
(WebKit::PlatformXRSystem::xrCoordinator):
* Source/WebKit/UIProcess/XR/openxr/PlatformXRSystemOpenXR.cpp:
(WebKit::PlatformXRSystem::xrCoordinator):
* Source/WebKit/UIProcess/ios/TextCheckerIOS.mm:
(WebKit::mutableState):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(isBuiltInScrollViewPanGestureRecognizer):
(isBuiltInScrollViewGestureRecognizer):
* Source/WebKit/UIProcess/ios/WKPageHostedModelView.mm:
(+[WKPageHostedModelView _usesStereoContent]):
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
(WebKit::desktopClassBrowsingSupported):
(WebKit::WebPageProxy::isDesktopClassBrowsingRecommended const):
* Source/WebKit/UIProcess/mac/LegacySessionStateCoding.cpp:
(WebKit::encodeSessionHistoryEntryData):
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::appUsesCustomAccentColor):
* Source/WebKit/UIProcess/mac/SecItemShimProxy.cpp:
(WebKit::SecItemShimProxy::singleton):
* Source/WebKit/UIProcess/mac/SecItemShimProxy.h:
* Source/WebKit/UIProcess/mac/ServicesController.mm:
(WebKit::ServicesController::refreshExistingServices):
* Source/WebKit/UIProcess/mac/TextCheckerMac.mm:
(WebKit::mutableState):
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::toolTipOwnerForSendingMouseEvents const):
* Source/WebKit/WebProcess/GPU/graphics/RemoteNativeImageProxy.cpp:
(WebKit::placeholderPlatformImage):
* Source/WebKit/WebProcess/GPU/media/AudioVideoRendererRemote.cpp:
(WebKit::AudioVideoRendererRemote::queueSingleton):
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
(WebKit::MediaSourcePrivateRemote::queueSingleton):
* Source/WebKit/WebProcess/cocoa/LaunchServicesDatabaseManager.mm:
(WebKit::LaunchServicesDatabaseManager::singleton):
* Source/WebKitLegacy/ios/Misc/WebGeolocationProviderIOS.mm:
(+[WebGeolocationProviderIOS sharedGeolocationProvider]):
* Source/WebKitLegacy/mac/Configurations/Base.xcconfig:
* Source/WebKitLegacy/mac/Misc/WebLocalizableStrings.mm:
* Source/WebKitLegacy/mac/WebCoreSupport/WebPlatformStrategies.mm:
(WebPlatformStrategies::initializeIfNecessary):

Canonical link: <a href="https://commits.webkit.org/303476@main">https://commits.webkit.org/303476@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f88791a3721e165fe74cff9050515b238e45337f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132569 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5064 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43631 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140088 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/84579 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/24a6c10e-4756-4ae4-9a74-1e58097dfd26) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134439 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/5244 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4823 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/101349 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68647 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/d2f3bb1e-b5d9-468a-bf27-f5ec73102965) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135515 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118757 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82146 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/81dd678b-c464-4d7d-ae51-a3b4e0a97192) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/3587 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/1346 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/83331 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/124639 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/112500 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36873 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142750 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/131077 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4734 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37461 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/110137 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4816 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4092 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/109913 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27845 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3604 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115029 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/58168 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4788 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33379 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/164044 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4624 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68239 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/42609 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4879 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4745 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WTF/Configurations/Base.xcconfig:
```diff
@@ -67,7 +67,7 @@ GCC_OBJC_CALL_CXX_CDTORS = YES;
 GCC_PREPROCESSOR_DEFINITIONS = $(DEBUG_DEFINES) $(inherited);
 GCC_STRICT_ALIASING = YES;
-GCC_THREADSAFE_STATICS = NO;
+GCC_THREADSAFE_STATICS = YES;
```
It would be slightly nicer to set this in Configurations/CommonBase.xcconfig and then un-set it in JavaScriptCore/Configurations/Base.xcconfig and bmalloc/Configurations/Base.xcconfig.

geoffreygaren reviewed (approved):
If you need a single-threaded static, is there a way to declare that?

cdumez committed:
Turn on thread-safe statics everywhere except bmalloc and JSC
https://bugs.webkit.org/show_bug.cgi?id=303020

Reviewed by Geoffrey Garen.

Turn on thread-safe statics everywhere except bmalloc and JSC. This
results in safer & simpler code without any noticeable performance
regression on the benchmarks we track.

* Configurations/CommonBase.xcconfig:
* Source/WTF/Configurations/Base.xcconfig:
* Source/WTF/wtf/Assertions.cpp:
* Source/WTF/wtf/CryptographicallyRandomNumber.cpp:
* Source/WTF/wtf/CurrentTime.cpp:
* Source/WTF/wtf/FastMalloc.cpp:
(WTF::MallocCallTracker::singleton):
* Source/WTF/wtf/Language.cpp:
(WTF::observerMap):
* Source/WTF/wtf/Logger.cpp:
(WTF::Logger::observers):
(WTF::Logger::messageHandlerObservers):
* Source/WTF/wtf/MainThreadDispatcher.cpp:
(WTF::MainThreadDispatcher::singleton):
* Source/WTF/wtf/MemoryPressureHandler.cpp:
(WTF::MemoryPressureHandler::singleton):
* Source/WTF/wtf/MemoryPressureHandler.h:
* Source/WTF/wtf/NumberOfCores.cpp:
(WTF::numberOfPhysicalProcessorCores):
* Source/WTF/wtf/OSRandomSource.cpp:
(WTF::cryptographicallyRandomValuesFromOS):
* Source/WTF/wtf/ParkingLot.cpp:
* Source/WTF/wtf/RAMSize.cpp:
(WTF::ramSize):
* Source/WTF/wtf/RunLoop.cpp:
(WTF::RunLoop::runLoopHolder):
* Source/WTF/wtf/SequesteredImmortalHeap.cpp:
(WTF::SequesteredImmortalHeap::instance):
* Source/WTF/wtf/StatisticsManager.cpp:
(WTF::StatisticsManager::singleton):
(WTF::StatisticsManager::~StatisticsManager):
* Source/WTF/wtf/StatisticsManager.h:
* Source/WTF/wtf/Threading.cpp:
(WTF::Thread::allThreads):
(WTF::Thread::initializeInThread):
* Source/WTF/wtf/TranslatedProcess.cpp:
(WTF::isX86BinaryRunningOnARM):
* Source/WTF/wtf/URL.cpp:
(WTF::aboutBlankURL):
(WTF::aboutSrcDocURL):
* Source/WTF/wtf/URLParser.cpp:
(WTF::URLParser::internationalDomainNameTranscoder):
* Source/WTF/wtf/UUID.cpp:
(WTF::bootSessionUUIDString):
* Source/WTF/wtf/WorkQueue.cpp:
(WTF::ConcurrentWorkQueue::apply):
(WTF::WorkQueue::mainSingleton):
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.mm:
(WTF::CocoaApplication::shouldOSFaultLogForAppleApplicationUsingWebKit1):
* Source/WTF/wtf/cocoa/SystemTracingCocoa.cpp:
(WTFSignpostLogHandle):
* Source/WTF/wtf/text/TextBreakIterator.cpp:
(WTF::TextBreakIteratorCache::singleton):
* Source/WTF/wtf/text/TextBreakIterator.h:
* Source/WTF/wtf/threads/Signals.cpp:
(WTF::activeThreads):
* Source/WTF/wtf/unicode/icu/CollatorICU.cpp:
(WTF::resolveDefaultLocale):
* Source/WebCore/Configurations/Base.xcconfig:
* Source/WebCore/Modules/applepay/ApplePayLogoSystemImage.mm:
(WebCore::passKitBundleSingleton):
(WebCore::applePayLogoWhite):
(WebCore::applePayLogoBlack):
* Source/WebCore/Modules/mediastream/RTCDataChannel.cpp:
(WebCore::WTF_REQUIRES_LOCK):
* Source/WebCore/Modules/system-preview/ARKitBadgeSystemImage.mm:
(WebCore::arKitBundle):
(WebCore::systemPreviewLogo):
* Source/WebCore/PAL/Configurations/Base.xcconfig:
* Source/WebCore/PAL/pal/cocoa/AVFoundationSoftLink.mm:
(PAL::AVFoundationLibrary):
* Source/WebCore/PAL/pal/system/ios/Device.cpp:
(PAL::deviceName):
* Source/WebCore/PAL/pal/text/EncodingTables.cpp:
(PAL::jis0208):
(PAL::jis0212):
(PAL::big5):
(PAL::gb18030):
* Source/WebCore/PAL/pal/text/TextCodecCJK.cpp:
(PAL::jis0208EncodeIndex):
(PAL::eucKREncodingIndex):
(PAL::big5EncodeIndex):
(PAL::gb18030EncodeIndex):
* Source/WebCore/PAL/pal/text/TextCodecSingleByte.cpp:
(PAL::tableForEncoding):
* Source/WebCore/SaferCPPExpectations/RetainPtrCtorAdoptCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedLambdaCapturesCheckerExpectations:
* Source/WebCore/bindings/js/WebCoreJSClientData.cpp:
(WebCore::JSHeapData::ensureHeapData):
* Source/WebCore/crypto/CryptoAlgorithmRegistry.cpp:
(WebCore::CryptoAlgorithmRegistry::singleton):
* Source/WebCore/crypto/CryptoAlgorithmRegistry.h:
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::maskedURLForBindings):
* Source/WebCore/editing/cocoa/DictionaryLookup.mm:
(WebCore::canCreateRevealItems):
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp:
(WebCore::activeContexts):
* Source/WebCore/html/track/TrackBase.cpp:
(WebCore::nullLogger):
* Source/WebCore/loader/ResourceMonitorThrottlerHolder.cpp:
(WebCore::ResourceMonitorThrottlerHolder::sharedWorkQueueSingleton):
* Source/WebCore/loader/cache/CachedResource.cpp:
(WebCore::CachedResource::response const):
(WebCore::CachedResource::resourceError const):
* Source/WebCore/page/NavigatorBase.cpp:
(WebCore::NavigatorBase::platform const):
(WebCore::NavigatorBase::hardwareConcurrency):
* Source/WebCore/page/NavigatorUAData.cpp:
(WebCore::NavigatorUAData::platform const):
* Source/WebCore/page/Performance.cpp:
(WebCore::isSignpostEnabled):
* Source/WebCore/page/cocoa/ResourceUsageOverlayCocoa.mm:
(WebCore::historicUsageData):
* Source/WebCore/page/linux/ResourceUsageThreadLinux.cpp:
(WebCore::threadInfoMap):
* Source/WebCore/page/scrolling/ScrollingThread.cpp:
(WebCore::ScrollingThread::singleton):
* Source/WebCore/page/scrolling/ScrollingThread.h:
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::webCoreBundleSingleton):
* Source/WebCore/platform/PublicSuffixStore.cpp:
(WebCore::PublicSuffixStore::singleton):
* Source/WebCore/platform/PublicSuffixStore.h:
* Source/WebCore/platform/cocoa/WebAVPlayerLayerView.mm:
(WebCore::allocWebAVPlayerLayerViewInstance):
(WebCore::allocWebAVPictureInPicturePlayerLayerViewInstance):
* Source/WebCore/platform/graphics/DestinationColorSpace.cpp:
(WebCore::knownColorSpace):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::globalLoaderDelegateQueue):
(WebCore::MediaPlayerPrivateAVFoundationObjC::asyncVideoPlaybackQualityMetrics):
* Source/WebCore/platform/graphics/avfoundation/objc/QueuedVideoOutput.mm:
(WebCore::globalOutputDelegateQueue):
* Source/WebCore/platform/graphics/cg/ColorCG.cpp:
(WebCore::cachedCGColorTransform):
(WebCore::cachedCGColor):
* Source/WebCore/platform/graphics/cocoa/WebActionDisablingCALayerDelegate.mm:
(+[WebActionDisablingCALayerDelegate shared]):
* Source/WebCore/platform/graphics/cocoa/WebMAudioUtilitiesCocoa.mm:
(WebCore::registerOpusDecoderIfNeeded):
* Source/WebCore/platform/graphics/gpu/cocoa/GPUAvailabilityMetal.mm:
(WebCore::isGPUAvailable):
* Source/WebCore/platform/ios/PlatformScreenIOS.mm:
(WebCore::screenPPIFactor):
* Source/WebCore/platform/ios/ValidationBubbleIOS.mm:
(allocWebValidationBubbleViewControllerInstance):
* Source/WebCore/platform/ios/VideoPresentationInterfaceAVKitLegacy.mm:
(allocWebAVPictureInPictureContentViewControllerInstance):
* Source/WebCore/platform/ios/WebItemProviderPasteboard.mm:
(+[WebItemProviderPasteboard sharedInstance]):
* Source/WebCore/workers/WorkerOrWorkletThread.cpp:
(WebCore::WorkerOrWorkletThread::workerOrWorkletThreads):
* Source/WebGPU/Configurations/Base.xcconfig:
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::Device::nopVertexFunction):
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
(WebGPU::ComputePassEncoder::runPredispatchIndirectCallValidation):
* Source/WebGPU/WebGPU/Device.mm:
* Source/WebGPU/WebGPU/HardwareCapabilities.mm:
(WebGPU::isShaderValidationEnabled):
(WebGPU::isPhysicalHardware):
* Source/WebInspectorUI/Configurations/Base.xcconfig:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.cpp:
(WebKit::remoteGraphicsContextGLStreamWorkQueueSingleton):
* Source/WebKit/GPUProcess/graphics/wc/RemoteWCLayerTreeHost.cpp:
(WebKit::remoteGraphicsStreamWorkQueue):
* Source/WebKit/GPUProcess/media/RemoteMediaResourceLoader.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
(WebKit::NetworkRTCProvider::rtcNetworkThread):
(WebKit::NetworkRTCProvider::createSocket): Deleted.
(WebKit::NetworkRTCProvider::signalSocketIsClosed): Deleted.
(WebKit::NetworkRTCProvider::protectedRTCMonitor): Deleted.
(WebKit::NetworkRTCProvider::sharedPreferencesForWebProcess): Deleted.
(WebKit::NetworkRTCProvider::updateSharedPreferencesForWebProcess): Deleted.
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCTCPSocketCocoa.mm:
(WebKit::tcpSocketQueueSingleton):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCUDPSocketCocoa.mm:
(WebKit::udpSocketQueueSingleton):
* Source/WebKit/Platform/IPC/cocoa/ConnectionCocoa.mm:
(IPC::shouldLogIncomingMessageHandling):
* Source/WebKit/Platform/classifier/cocoa/ResourceLoadStatisticsClassifierCocoa.cpp:
(WebKit::ResourceLoadStatisticsClassifierCocoa::singletonPredictionModel):
* Source/WebKit/Platform/cocoa/ImageAnalysisUtilities.mm:
(WebKit::shouldLogFullImageTranslationResults):
* Source/WebKit/Shared/API/APIObject.cpp:
(API::Object::apiObjectsUnderConstruction):
* Source/WebKit/Shared/Cocoa/CompletionHandlerCallChecker.mm:
(WebKit::shouldThrowExceptionForDuplicateCompletionHandlerCall):
* Source/WebKit/Shared/Cocoa/CoreIPCSecureCoding.mm:
(WebKit::SecureCoding::internalClassNamesExemptFromSecureCodingCrash):
* Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm:
(WebKit::isInWebKitChildProcess):
(WebKit::doesParentProcessHaveTrackingPreventionEnabled):
(WebKit::isParentProcessAFullWebBrowser):
(WebKit::isFullWebBrowserOrRunningTest):
* Source/WebKit/Shared/Cocoa/LaunchLogHook.h:
* Source/WebKit/Shared/Cocoa/LaunchLogHook.mm:
(WebKit::LaunchLogHook::singleton):
* Source/WebKit/Shared/Cocoa/WebPreferencesDefaultValuesCocoa.mm:
(WebKit::defaultTextRecognitionInVideosEnabled):
(WebKit::defaultVisualTranslationEnabled):
(WebKit::defaultRemoveBackgroundEnabled):
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.mm:
(WebKit::setOSTransaction):
* Source/WebKit/Shared/Extensions/WebExtensionUtilities.mm:
(WebKit::classToClassString):
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultMediaSessionCoordinatorEnabled):
(WebKit::defaultDigitalCredentialsEnabled):
* Source/WebKit/Shared/ios/WebPreferencesDefaultValuesIOS.mm:
(WebKit::isAsyncTextInputFeatureFlagEnabled):
* Source/WebKit/Shared/unix/BreakpadExceptionHandler.cpp:
(WebKit::installBreakpadExceptionHandler):
* Source/WebKit/UIProcess/API/Cocoa/APIContentRuleListStoreCocoa.mm:
(API::ContentRuleListStore::defaultStorePath):
* Source/WebKit/UIProcess/API/Cocoa/WKNSURLAuthenticationChallenge.mm:
(-[WKNSURLAuthenticationChallenge _web_createTarget]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStore.mm:
(+[WKWebsiteDataStore allWebsiteDataTypes]):
(+[WKWebsiteDataStore _allWebsiteDataTypesIncludingPrivate]):
* Source/WebKit/UIProcess/Cocoa/MediaPermissionUtilities.mm:
(WebKit::checkSandboxRequirementForType):
(WebKit::checkUsageDescriptionStringForType):
* Source/WebKit/UIProcess/Cocoa/UIProcessLogInitializationCocoa.mm:
(WebKit::UIProcess::wtfLogLevelString):
(WebKit::UIProcess::webCoreLogLevelString):
(WebKit::UIProcess::webKitLogLevelString):
* Source/WebKit/UIProcess/Cocoa/WebKitSwiftSoftLink.mm:
(WebKit::WebKitSwiftLibrary):
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::WebProcessPool::setMediaAccessibilityPreferences):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCommandCocoa.mm:
(WebKit::WebExtensionCommand::setActivationKey):
* Source/WebKit/UIProcess/Extensions/Cocoa/_WKWebExtensionDeclarativeNetRequestRule.mm:
(validRequestMethodsString):
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::ProcessLauncher::hasExtensionsInAppBundle):
* Source/WebKit/UIProcess/Media/cocoa/MediaUsageManagerCocoa.mm:
(WebKit::usageTrackingAvailable):
* Source/WebKit/UIProcess/Notifications/glib/NotificationService.cpp:
(WebKit::NotificationService::singleton):
* Source/WebKit/UIProcess/Notifications/glib/NotificationService.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeViews.mm:
(WebKit::scrollViewScrollIndicatorClassSingleton):
* Source/WebKit/UIProcess/SuspendedPageProxy.cpp:
(WebKit::messageNamesToIgnoreWhileSuspended):
* Source/WebKit/UIProcess/WebPasteboardProxy.cpp:
(WebKit::WebPasteboardProxy::singleton):
* Source/WebKit/UIProcess/WebPasteboardProxy.h:
* Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm:
(WebKit::defaultWebsiteDataStoreRootDirectory):
(WebKit::WebsiteDataStore::tempDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::cacheDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::websiteDataDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::defaultResolvedContainerTemporaryDirectory):
* Source/WebKit/UIProcess/XR/ios/PlatformXRSystemIOS.mm:
(WebKit::PlatformXRSystem::xrCoordinator):
* Source/WebKit/UIProcess/XR/openxr/PlatformXRSystemOpenXR.cpp:
(WebKit::PlatformXRSystem::xrCoordinator):
* Source/WebKit/UIProcess/ios/TextCheckerIOS.mm:
(WebKit::mutableState):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(isBuiltInScrollViewPanGestureRecognizer):
(isBuiltInScrollViewGestureRecognizer):
* Source/WebKit/UIProcess/ios/WKPageHostedModelView.mm:
(+[WKPageHostedModelView _usesStereoContent]):
* Source/WebKit/UIProcess/ios/WebPageProxyIOS.mm:
(WebKit::desktopClassBrowsingSupported):
(WebKit::WebPageProxy::isDesktopClassBrowsingRecommended const):
* Source/WebKit/UIProcess/mac/LegacySessionStateCoding.cpp:
(WebKit::encodeSessionHistoryEntryData):
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::appUsesCustomAccentColor):
* Source/WebKit/UIProcess/mac/SecItemShimProxy.cpp:
(WebKit::SecItemShimProxy::singleton):
* Source/WebKit/UIProcess/mac/SecItemShimProxy.h:
* Source/WebKit/UIProcess/mac/ServicesController.mm:
(WebKit::ServicesController::refreshExistingServices):
* Source/WebKit/UIProcess/mac/TextCheckerMac.mm:
(WebKit::mutableState):
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::toolTipOwnerForSendingMouseEvents const):
* Source/WebKit/WebProcess/GPU/graphics/RemoteNativeImageProxy.cpp:
(WebKit::placeholderPlatformImage):
* Source/WebKit/WebProcess/GPU/media/AudioVideoRendererRemote.cpp:
(WebKit::AudioVideoRendererRemote::queueSingleton):
* Source/WebKit/WebProcess/GPU/media/MediaSourcePrivateRemote.cpp:
(WebKit::MediaSourcePrivateRemote::queueSingleton):
* Source/WebKit/WebProcess/cocoa/LaunchServicesDatabaseManager.mm:
(WebKit::LaunchServicesDatabaseManager::singleton):
* Source/WebKitLegacy/ios/Misc/WebGeolocationProviderIOS.mm:
(+[WebGeolocationProviderIOS sharedGeolocationProvider]):
* Source/WebKitLegacy/mac/Configurations/Base.xcconfig:
* Source/WebKitLegacy/mac/Misc/WebLocalizableStrings.mm:
* Source/WebKitLegacy/mac/WebCoreSupport/WebPlatformStrategies.mm:
(WebPlatformStrategies::initializeIfNecessary):

Canonical link: https://commits.webkit.org/303476@main
